alacritty: 0.12.3 -> 0.13.0

r-vdp opened the pull request:
## Description of changes

https://github.com/alacritty/alacritty/releases/tag/v0.13.0

We should wait for https://github.com/nix-community/home-manager/pull/4817 to be merged to avoid breaking the home-manager module.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-vdp committed:
alacritty: 0.12.3 -> 0.13.0

wahjava commented:
Result of `nixpkgs-review pr 277316` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>alacritty</li>
    <li>alacritty.terminfo</li>
    <li>wio</li>
  </ul>
</details>

tukanoidd commented:
Any news? The PR in HM got merged

nicoonoclaste committed:
alacritty-theme: unstable-2023-11-07 → unstable-2023-12-28

Necessary for alacritty v0.13, as the configuration language changed (TOML instead of YAML)

r-vdp commented:
@nbraud cool, thanks! I merged it into this branch.

nicoonoclaste reviewed (approved):
LGTM, overall.  I'm not super keen on `postPatch`'s complexity, but it doesn't seem possible (yet?) to leverage upstream's `Makefile` instead.

The suggestion is just that, I'm OK with whichever version the drv's maintainers prefer.

r-vdp commented on pkgs/applications/terminal-emulators/alacritty/default.nix:
```diff
@@ -107,16 +108,17 @@ rustPlatform.buildRustPackage rec {
+    scdoc < extra/man/alacritty.1.scd | gzip -c > $out/share/man/man1/alacritty.1.gz
+    scdoc < extra/man/alacritty-msg.1.scd | gzip -c > $out/share/man/man1/alacritty-msg.1.gz
+    scdoc < extra/man/alacritty.5.scd | gzip -c > $out/share/man/man5/alacritty.5.gz
+    scdoc < extra/man/alacritty-bindings.5.scd | gzip -c > $out/share/man/man5/alacritty-bindings.5.gz
```
Yeah, we could make a PR upstream for this, it seems very sensible that they handle this complexity so that it can be reused by everyone.

r-vdp replied:
I think I prefer the current approach which doesn't use a bunch of cryptic bash-isms, especially since it's only 4 lines.
I think the best way forward to avoid this complexity, would be an upstream PR.

r-vdp commented:
> Thanks a bunch!  FWIW, you could have used a fast-forward merge (`git merge --ff`) to avoid introducing extraneous merge commits into the history, but you can still get rid of it by rebasing.

Yeah, I'm on mobile and I clicked the merge button too quickly. I rebased the branch in the meantime.

nicoonoclaste reviewed (approved):
Still LGTM

vcunat commented:
Since this commit updating `alacritty`, the `nixosTests.sway` loops (bisected).  That currently blocks the nixos-unstable channel.  Example log: https://hydra.nixos.org/build/245727011/nixlog/20

/cc maintainers of the test: @primeos, @Synthetica9

Mic92 commented:
Breaks on this line: https://github.com/NixOS/nixpkgs/blob/868769dc6c52c597dcf8275cd8eb7c019787536e/nixos/tests/sway.nix#L138

trofi commented:
Bisect says 1fdcdc2f1be30b500145ea74a7aec7df36062390 `alacritty-theme: unstable-2023-11-07 → unstable-2023-12-28` broke the `atacritty-theme` build on `x86_64-linux` as:

```
$ nix build --no-link -f. alacritty-theme -L
alacritty-theme> Running phase: unpackPhase
alacritty-theme> unpacking source archive /nix/store/fspc2v4jrpfx30avp5aagfigq0vhy8pz-source
alacritty-theme> source root is source/themes
alacritty-theme> Running phase: patchPhase
alacritty-theme> Running phase: updateAutotoolsGnuConfigScriptsPhase
alacritty-theme> Running phase: installPhase
alacritty-theme> install: missing file operand
alacritty-theme> Try 'install --help' for more information.
```

trofi commented:
Proposed the fix as https://github.com/NixOS/nixpkgs/pull/279354

r-vdp commented:
Yeah they changed this upstream, thanks for the fix!
